src/main/java/com/exasol/glue/connection/ExasolConnectionFactory.java

```diff
@@ -0,0 +1,74 @@
+package com.exasol.glue.connection;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.logging.Logger;
+
+import com.exasol.glue.ExasolOptions;
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * A factory that creates JDBC connection to Exasol database.
+ */
+public final class ExasolConnectionFactory {
+    private static final Logger LOGGER = Logger.getLogger(ExasolConnectionFactory.class.getName());
+    private final ExasolOptions options;
+    private Connection connection = null;
+
+    /**
+     * Creates an instance of a {@link ExasolConnectionFactory}.
+     *
+     * @param options an {@link ExasolOptions} options
+     */
+    public ExasolConnectionFactory(final ExasolOptions options) {
+        this.options = options;
+    }
+
+    /**
+     * Creates a JDBC connection to an Exasol database.
+     *
+     * @return JDBC connection
+     * @throws SQLException if the connection cannot be established
+     */
+    public synchronized Connection getConnection() throws SQLException {
+        if (this.connection == null) {
+            this.connection = createConnection();
+        }
+        return this.connection;
```

Comment:
What if the user closes the connection? Then you have no way of creating a new one.
Maybe don't cache the connection here and instead pass it explicitly.

Reply:
That is good point. At the moment the caching is not needed at all.